packages/flutter/lib/src/widgets/list_wheel_scroll_view.dart

```diff
@@ -1,15 +1,338 @@
+    return new FixedExtentScrollPhysics(parent: buildParent(ancestor));
+  }
+
+  @override
+  Simulation createBallisticSimulation(ScrollMetrics position, double velocity) {
+    assert(
+      position is _FixedExtentScrollPosition,
+      'FixedExtentScrollPhysics can only be used with Scrollables that uses '
+      'the FixedExtentScrollController'
+    );
+
+    final _FixedExtentScrollPosition metrics = position;
+
+    // Scenario 1:
+    // If we're out of range and not headed back in range, defer to the parent
+    // ballistics, which should put us back in range at the scrollable's boundary.
+    if ((velocity <= 0.0 && metrics.pixels <= metrics.minScrollExtent) ||
+        (velocity >= 0.0 && metrics.pixels >= metrics.maxScrollExtent)) {
+      return super.createBallisticSimulation(metrics, velocity);
+    }
+
+    // Create a test simulation to see where it would have ballistically fallen
+    // naturally without settling onto items.
+    final FrictionSimulation testFrictionSimulation =
+        // 0.135 is an arbitrary number copied from scroll_simulation.
+        new FrictionSimulation(0.135, metrics.pixels, velocity);
+
+    // Scenario 2:
+    // If it was going to end up past the scroll extent, defer back to the
+    // parent physics' ballistics again which should put us on the scrollable's
+    // boundary.
+    if (testFrictionSimulation.finalX < metrics.minScrollExtent
+        || testFrictionSimulation.finalX > metrics.maxScrollExtent) {
+      return super.createBallisticSimulation(metrics, velocity);
+    }
+
+    // From the natural final position, find the nearest item it should have
+    // settled to.
+    final int settlingItemIndex = _getItemFromOffset(
+      offset: testFrictionSimulation.finalX,
+      itemExtent: metrics.itemExtent,
+      minScrollExtent: metrics.minScrollExtent,
+      maxScrollExtent: metrics.maxScrollExtent,
+    );
+
+    final double settlingPixels = settlingItemIndex * metrics.itemExtent;
+
+    // Scenario 3:
+    // If there's no velocity and we're already at where we intend to land,
+    // do nothing.
+    if (velocity.abs() < tolerance.velocity
+        && (settlingPixels - metrics.pixels).abs() < tolerance.distance) {
+      return null;
+    }
+
+    // Scenario 4:
+    // If we're going to end back at the same item because initial velocity
+    // is too low to break past it, use a spring simulation to get back.
+    if (settlingItemIndex == metrics.itemIndex) {
+      return new SpringSimulation(
+        spring,
+        metrics.pixels,
+        settlingPixels,
+        velocity,
+        tolerance: tolerance,
+      );
+    }
+
+    // Scenario 5:
+    // Create a new friction simulation except the drag will be tweaked to land
+    // exactly on the item closest to the natural stopping point.
+    return new FrictionSimulation.through(
+      metrics.pixels,
+      settlingPixels,
+      velocity,
+      tolerance.velocity * velocity.sign,
+    );
+  }
+}
```

Comment:
how similar to iOS does this feel?

A possibly more realistic (maybe less like iOS?) physics model would be to model each item as a frictive potential valley, essentially a series of back-to-back springs where if you cross the threshold of one spring you enter the next with velocity. That way if you fling it to the next item with enough force to overshoot but not enough to reach the next item, you'd see it go forward then spring back, rathen that land perfectly on it.

But we should match iOS.